ibis/config.py

```diff
@@ -172,6 +172,9 @@ class Options(Config):
 
 
 def _default_backend() -> Any:
+    if (backend := options.default_backend) is not None:
+        return backend
```

Comment:
Should we maybe move to storing the duckdb con on `options.default_backend` rather than a hidden global variable?

Reply:
I guess as written the user can reset `options.default_backend = None` later on and the same duckdb connection will be reused, rather than opening a new one.

Reply:
The idea was to reuse the existing in-memory duckdb connection if it's ever created.

Reply:
right, but it could be reused if it's set on `options.default_backend` instead of a hidden global.

The only difference is that with the current implementation the following works:

```python
con1 = _default_backend()  # a duckdb backend

# Manually set the default backend to something else
con2 = ibis.bigquery.connect(...)
options.default_backend = con2
assert _default_backend() is con2

# Reset it back to None
options.default_backend = None

con3 = _default_backend()
assert con1 is con3  # original duckdb backend is reused
```

If this is intended behavior, then what you have here seems fine. No strong thoughts.

Reply:
I see what you're saying. It wasn't intentional when I wrote this originally. I think it probably makes more sense for `None` to reset the default backend rather than have a global floating around.